spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/MilliSpan.java

```diff
@@ -35,10 +35,10 @@
 	private final long begin;
 	private long end = 0;
 	private final String name;
-	private final String traceId;
+	private final long traceId;
 	@Singular
-	private List<String> parents = new ArrayList<>();
-	private final String spanId;
+	private List<Long> parents = new ArrayList<>();
+	private final Long spanId;
```

Comment:
Yes, if you only have log based tracing then there doesn't have to be a notion of spans